crates/nu-command/src/strings/str_/replace.rs

```diff
@@ -91,7 +96,19 @@ impl Command for SubCommand {
         let cell_paths: Vec<CellPath> = call.rest(engine_state, stack, 2)?;
         let cell_paths = (!cell_paths.is_empty()).then_some(cell_paths);
         let literal_replace = call.has_flag("no-expand");
-        let no_regex = call.has_flag("string");
+        if call.has_flag("string") {
+            report_error_new(
+                engine_state,
+                &ShellError::GenericError(
```

Comment:
If we're throwing an error here, it's not really deprecated, right? It may as well be removed if we cannot use it.

Reply:
i do not think this is throwing an error, right?

the `GenericError` is passed to `report_error_new` which should only print the error, as in https://github.com/nushell/nushell/pull/10017 :yum: 
https://github.com/nushell/nushell/blob/8c52b7a23a880f4c7ff1b55be84cf3e115cf1356/crates/nu-protocol/src/cli_error.rs#L33-L40

Reply:
Oh, if it's still allowing you to use it, then it's deprecated correctly. Although that's friggin annoying. LOL
```nushell
❯ (pwd) | str replace nushell nush -s
Error:   × Deprecated option
   ╭─[entry #1:1:1]
 1 │ (pwd) | str replace nushell nush -s
   ·         ─────┬─────
   ·              ╰── `str replace --string` is deprecated and will be removed in 0.85.
   ╰────
  help: Substring matching is now the default. Use `--regex` or `--multiline` for matching regular expressions.


C:\Users\dschroeder\source\repos\forks\nush
```

Reply:
ohh yeah it will be until users remove their calls to this option :laughing:

Reply:
feel free to close this thread if it's ok with you :wink:

Reply:
Yeah, it will only print the error but should keep working.